spec/support/usps_ipp_helper.rb

```diff
@@ -238,4 +267,11 @@ def stub_request_enrollment_code
       headers: { 'content-type' => 'application/json' },
     )
   end
+
+  def stub_request_enrollment_code_with_forbidden_error
+    stub_request(
+      :post,
+      %r{/ivs-ippaas-api/IPPRest/resources/rest/requestEnrollmentCode},
+    ).to_raise(Faraday::ForbiddenError)
```

Comment:
Quick question, would Faraday still return the same error if we had something like `to_return(status: 403)`
```suggestion
    ).to_return(status: 403, body: '')
```

Reply:
It's slightly different.  Here's what I saw, using binding.pry, for the `raise` option.

```
 #<WebMock::RequestStub:0x000000011abda978
 @request_pattern=
  #<WebMock::RequestPattern:0x000000011abda928
   @body_pattern=nil,
   @headers_pattern=nil,
   @method_pattern=#<WebMock::MethodPattern:0x000000011abda8d8 @pattern=:post>,
   @uri_pattern=
    #<WebMock::URIRegexpPattern:0x000000011abda888
     @pattern=/\/ivs-ippaas-api\/IPPRest\/resources\/rest\/requestEnrollmentCode/,
     @query_params=nil>,
   @with_block=nil>,
 @responses_sequences=
  [#<WebMock::ResponsesSequence:0x000000011abda5b8
    @current_position=0,
    @responses=
     [#<WebMock::Response:0x000000011abda798
       @body=nil,
       @exception=#<Faraday::ForbiddenError #<Faraday::ForbiddenError: Exception from WebMock>>,
       @headers=nil,
       @should_timeout=nil,
       @status=nil>],
    @times_to_repeat=1>]>
```

Here's what I see with the return option.

```
#<WebMock::RequestStub:0x000000010e80a840
 @request_pattern=
  #<WebMock::RequestPattern:0x000000010e80a7f0
   @body_pattern=nil,
   @headers_pattern=nil,
   @method_pattern=#<WebMock::MethodPattern:0x000000010e80a7a0 @pattern=:post>,
   @uri_pattern=
    #<WebMock::URIRegexpPattern:0x000000010e80a750
     @pattern=/\/ivs-ippaas-api\/IPPRest\/resources\/rest\/requestEnrollmentCode/,
     @query_params=nil>,
   @with_block=nil>,
 @responses_sequences=
  [#<WebMock::ResponsesSequence:0x000000010e80a4d0
    @current_position=0,
    @responses=
     [#<WebMock::Response:0x000000010e80a660
       @body="",
       @exception=nil,
       @headers=nil,
       @should_timeout=nil,
       @status=[403, ""]>],
    @times_to_repeat=1>]>
```

Changing the stub does not change test behavior.

Reply:
I have a slight preference for having the mock server return a response, and letting the adapter translate + throw that, but since it doesn't change the tests, its prob fine